tests/tests/fee_tests.cpp

```diff
@@ -213,6 +213,102 @@ BOOST_AUTO_TEST_CASE(asset_claim_fees_test)
    FC_LOG_AND_RETHROW()
 }
 
+BOOST_AUTO_TEST_CASE(asset_claim_pool_test)
+{
+    try
+    {
+        ACTORS((alice)(bob));
+        // Alice and Bob create some user issued assets
+        // Alice deposits BTS to the fee pool
+        // Alice claimes fee pool of her asset and can't claim pool of Bob's asset
+
+        const share_type core_prec = asset::scaled_precision( asset_id_type()(db).precision );
+
+        // return number of core shares (times precision)
+        auto _core = [&core_prec]( int64_t x ) -> asset
+        {  return asset( x*core_prec );    };
+
+        const asset_object& alicecoin = create_user_issued_asset( "ALICECOIN", alice,  0 );
+        const asset_object& aliceusd = create_user_issued_asset( "ALICEUSD", alice, 0 );
+        asset_id_type bobcoin_id = create_user_issued_asset( "BOBCOIN", bob, 0).id;
+
+        // prepare users' balance
+        issue_uia( alice, aliceusd.amount( 20000000 ) );
+        issue_uia( alice, alicecoin.amount( 10000000 ) );
+
+        transfer( committee_account, alice_id, _core(1000) );
+        transfer( committee_account, bob_id, _core(1000) );
+
+        enable_fees();
+
+        auto claim_pool = [&]( const account_id_type& issuer, const asset_id_type& asset_to_claim,
+                              const asset& amount_to_fund, const asset_object& fee_asset  )
+        {
+            asset_claim_pool_operation claim_op;
+            claim_op.issuer = issuer;
+            claim_op.asset_id = asset_to_claim;
+            claim_op.amount_to_claim = amount_to_fund;
+
+            signed_transaction tx;
+            tx.operations.push_back( claim_op );
+            db.current_fee_schedule().set_fee( tx.operations.back(), fee_asset.options.core_exchange_rate );
+            set_expiration( db, tx );
+            sign( tx, alice_private_key );
+            PUSH_TX( db, tx );
+
+        };
+
+        const asset_object& core_asset = asset_id_type()(db);
+
+        if( db.head_block_time() <= HARDFORK_CORE_188_TIME )
+        {
+           // can't claim pool before hardfork
+           GRAPHENE_REQUIRE_THROW( claim_pool( alice_id, aliceusd.id, _core(200), core_asset), fc::exception );
```

Comment:
You said:
> that test fails because we are befor the hardfork.

No. If an exception is caught here, it's also possible that it's due to insufficient fund in the pool. Say, if you remove [this line](https://github.com/bitshares/bitshares-core/blob/09969d50c5c891d52c73355809bc0138dbd23d54/libraries/chain/asset_evaluator.cpp#L627) (to create a obvious bug), the test will pass too, which means the unit test is not well designed. To test the hard fork time logic, you should design a case that no exception will be thrown out due to other reason.

Another thing, as @pmconrad mentioned in another PR, test case should be deterministic, so `if( db.head_block_time() <= HARDFORK_CORE_188_TIME )` doesn't make sense here.

Reply:
Understood ... and fixed (hopefully)